src/PanelTraits/AutoSet.php

```diff
@@ -36,8 +63,31 @@ public function setFromDb()
                     'label' => ucfirst($field),
                     'type'  => $this->getFieldTypeFromDbColumnType($field),
                 ];
+
+                if(!is_null($crudTypes) && isset($crudTypes[$field])) {
+                    if(is_array($crudTypes[$field])) {
+                        foreach($crudTypes[$field] as $key => $value) {
+                            $this->columns[$field][$key] = $value;
+                        }
+                    } else {
+                        $this->columns[$field]['type'] = $crudTypes[$field];
+                    }
+                }
             }
+
         }, $this->getDbColumnsNames());
+
+        if(!is_null($crudFakeFields)) {
+            foreach($crudFakeFields as $key => $data) {
+                $data['name'] = $key;
```

Comment:
I can see why this is being done but it's still somewhat yucky (i.e. we're not really morphing the `$data` variable but it looks like it).